src/wp-includes/functions.php

```diff
@@ -824,7 +824,7 @@ function wp_extract_urls( $content ) {
 			. '(?:'
 				. '\([\w\d]+\)|'
 				. '(?:'
-					. "[^`!()\[\]{};:'\".,<>«»“”‘’\s]|"
+					. "[^`!()\[\]{}:'\".,<>«»“”‘’\s]|"
```

Comment:
I'm concerned about the backwards-compatibility break being introduced here. Any URLs that contain a `;` will break as the `;` will be retained.

For example for `https://example.org;`:
- Current behavior: `https://example.org` is returned
- With this change: `https://example.org;` is returned
 
This change changes these URLs. While the change fixes the original reported issue, it causes another.

Instead, I'd suggest identifying and excluding only the `&amp;` pattern. Makes the regex more complicated. But retains back-compat and fixes the reported issue.